Packs/ZoomMail/Integrations/ZoomMail/ZoomMail.py

```diff
@@ -0,0 +1,1224 @@
+from traceback import format_exc
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+import json
+import base64
+from dateparser import parse
+from datetime import datetime, timedelta
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.mime.base import MIMEBase
+from email import encoders
+
+from collections.abc import Callable
+
+ZOOM_MAIL_COMMAND_PREFIX = "zoom-mail"
+
+
+class ZoomMailClient(BaseClient):
+    def __init__(
+        self,
+        base_url,
+        client_id,
+        client_secret,
+        account_id,
+        default_email,
+        verify=True,
+        proxy=False,
+    ):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.account_id = account_id
+        self.access_token = None
+        self.token_time: float = 0
+        self.default_email = default_email  # Default email added here
+
+    def obtain_access_token(self):
+        """
+        Obtains an access token using the 'account_credentials' grant type.
+        """
+        client_credentials = base64.b64encode(
+            f"{self.client_id}:{self.client_secret}".encode()
+        ).decode()
+
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+            "Authorization": f"Basic {client_credentials}",
+        }
+
+        body = {"grant_type": "account_credentials", "account_id": self.account_id}
+
+        response = super()._http_request(
+            method="POST",
+            full_url="https://zoom.us/oauth/token",
+            headers=headers,
+            data=body,
+        )
+
+        access_token = response.get("access_token")
+        if access_token:
+            self.access_token = access_token
+            self.token_time = time.time()
+            return {"success": True, "token": access_token}
+
+        return {
+            "success": False,
+            "error": "Failed to retrieve access token from ZoomMail API",
+        }
+
+    def _http_request(self, *args, **kwargs):
+        """
+        Override the _http_request method to include the access token in the headers.
+        """
+        if not self.access_token or (time.time() - self.token_time) >= 3500:
```

Comment:
if access token is not saved to integration context, what of saving and checking its creation time?
BTW, I don't  think we should save it, unless there are relevant limitations in the docs.

Reply:
I don't think it's being saved so much as we are holding it in memory as part of the ZoomMail class. I do think the time comparison we can remove however since we aren't storing it.

Reply:
ok, so lets remove.
No variable shed be saved on the container between runs.

Reply:
Actually re-reading this, I think it should stay. It's not being persisted to context or anything so the credentials are safe, just held in memory for the client. When the container dies, so does the client. If a call takes longer then the access token could theoretically expire before it's used again which could be useful for lists where the container is kept alive.